File: problem5/vers_cpp/p5.cpp
/*
 * p5.cpp
 *
 * Zeke Reyna
 *
 * Finds smallest int that is divisible by natural nums up to bound
 */

#include <iostream>
#define BOUND 10
using namespace std;

int main(int argc, char **argv)
{
	int i, num, bound;
	int isDivis;

	(argc == 2) ? bound = stoi(argv[1]) : bound = BOUND;

	num = 3;
	isDivis = 0;
	while (!isDivis && num++) {
		isDivis = 1;
		for (i = 1; i <= bound; i++) {
			if (num % i != 0) {
				isDivis = 0;
				break;
			}
		}
	}

	cout << "\nThe smallest number that is evenly divisible by all of ";
	cout << "the numbers from 1 to " << bound;
	cout << " is " << num << "\n\n";
	return 0;
}

File: problem5/vers_java/P5.java
public class P5
{
        public static void print( Object line )
        {
                System.out.print( line );
        }

        public static void main( String args[] )
        {
                int i, num, bound;
                boolean isDivis;

                if( args.length == 1 )
                        bound = Integer.parseInt( args[0] );
                else
                        bound = 10;

                num = 3;
                isDivis = false;
                while( !isDivis ) {
                        isDivis = true;
                        num++;
                        for( i = 1; i <= bound; i++ ) {
                                if( num % i != 0 ) {
                                        isDivis = false;
                                        break;
                                }
                        }
                }
                print("\nThe smallest number that is evenly divisible by all of");
                print(" the numbers from 1 to "+bound+" is "+num+"\n\n");
        }
}
